### ISSUE TITLE
Metric Concern: Evaluation Protocol Mismatch with SwinDR (Object vs. Valid Pixels)

### ISSUE THREAD

**Issue by rcao-hk** [songlin/d3roma#22]
<p>Hi, thanks for the great work on <strong>D3RoMa</strong>!<br>
While reproducing the reported results, I noticed a <strong>metric mismatch</strong> between the paper and the released evaluation script.</p>
<hr>
<h3> Reproduction Summary</h3>
<p>I re-trained on <strong>DREDS-CatKnown</strong> using <code inline="">train_dreds_reprod.yaml</code> and the legacy inference code:</p>
<pre><code class="language-python">from core.custom_pipelines import GuidedDiffusionPipeline, GuidedLatentDiffusionPipeline

clazz_pipeline = GuidedLatentDiffusionPipeline if config.ldm else GuidedDiffusionPipeline
pipeline = clazz_pipeline.from_pretrained(pretrained_path).to("cuda")
pipeline.guidance.flow_guidance_mode = config.flow_guidance_mode
</code></pre>
<p>When testing, the repo’s evaluation script computes metrics <strong>on all valid pixels</strong>:<br>
<a href="https://github.com/songlin/d3roma/blob/33130323ea771078377ef85000a8ccd10504e9a0/evaluate.py#L53C30-L53C34">https://github.com/songlin/d3roma/blob/33130323ea771078377ef85000a8ccd10504e9a0/evaluate.py#L53C30-L53C34</a></p>

<p>However, the paper states that it <em>“follows SwinDR-Net metrics”</em>,<br>
which evaluate <strong>only on object pixels</strong> (foreground masks).</p>
<hr>
<h3>Comparison Results</h3>

From | Dataset | A1 ↑ | A2 ↑ | A3 ↑ | RMSE ↓ | AbsRel ↓ | MAE ↓
-- | -- | -- | -- | -- | -- | -- | --
DRNet (report, object) | DREDS-CatKnown | 0.9804 | 0.9962 | 0.9998 | 0.0100 | 0.0080 | 0.0050
D3RoMa (report, all) | DREDS-CatKnown | 0.9964 | 0.9988 | 0.9999 | 0.0045 | 0.0016 | 0.0011
D3RoMa (reproduce, all) | DREDS-CatKnown | 0.9965 | 0.9989 | 0.9999 | 0.0045 | 0.0018 | 0.0013
D3RoMa (reproduce, object) | DREDS-CatKnown | 0.9811 | 0.9949 | 0.9997 | 0.0101 | 0.0075 | 0.0050
DRNet (report, object) | DREDS-CatNovel | 0.8189 | 0.9218 | 0.9839 | 0.0215 | 0.0336 | 0.0128
D3RoMa (reproduce, all) | DREDS-CatNovel | 0.9445 | 0.9706 | 0.9932 | 0.0142 | 0.0124 | 0.0053
D3RoMa (reproduce, object) | DREDS-CatNovel | 0.7957 | 0.8894 | 0.9829 | 0.0247 | 0.0358 | 0.0141

<hr>
<h3> Question</h3>
<p>Could you clarify whether the paper’s results were computed<br> <strong>on object pixels</strong> (as in SwinDR) or <strong>on all valid pixels</strong> (as in the released repo)?</p>
<p>This difference significantly affects the absolute metric values and fair comparison with other methods.</p>
<p>Thanks again for your contribution.

Best,

**Comment by songlin** [comment]
@rcao-hk Thanks for asking. I think we reported the results as in the released repo.  We didnot apply any object masks. I also want to clarify that the paper states that it “follows SwinDR-Net metrics”, it means we report same kind of metrics calculations as DREDS.
